### PR TITLE
Test commit

### PR DESCRIPTION
**Pull request by cs-tushar-kanade**

### CODE REVIEW

File: connector.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        scenarios = next(os.walk(os.path.join(os.path.dirname(__file__),"scenarios")))[1]
+        
+        if(config.get('threatIntel')):
+          load_threat()
+        
+        
+        if(config.get('import_records')):
+          for scenario in scenarios:
+              record_data_file = os.path.join(os.path.dirname(__file__),"scenarios/"+scenario+"/scenario_record.json")
+              logger.warn(record_data_file)
```
Comment (cs-pooja-singh) [inline connector.py]: logger.debug

File: info.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
+{
+    "name": "fortisoar-soc-simulator_dev",
```
Comment (cs-pooja-singh) [inline info.json]: fortisoar-soc-simulator

File: info.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    "version": "1.0.6",
+    "label": "FortiSOAR SOC Simulator",
+    "description": "FortiSOAR SOC Simulator",
+    "publisher": "FortiSOAR",
+    "icon_small_name": "fortisoar_small.png",
+    "icon_large_name": "fortisoar_large.png",
+    "category": "information",
+    "icon_medium_name": {
```
Comment (cs-pooja-singh) [inline info.json]: this should point to the image file

File: info.json [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            "title": "Bad IP",
+            "category": "investigation",
+            "annotation": "bad_ip",
+            "description": "Get Known bad IP",
+            "enabled": true,
+            "output_schema": {},
+            "parameters": [
+                {
+                    "title": "Random",
+                    "description": "Check to generate random Bad IP",
```
Comment (cs-pooja-singh) [inline info.json]: please add a tooltip as well. description will show up in the doc only

File: info.json [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                    "editable": true,
+                    "visible": true,
+                    "type": "textarea",
+                    "name": "variables",
+                    "value": ""
+                }
+            ]
+        }
+    ],
+    "forked_from": "fortisoar-soc-simulator_1_0_5",
```
Comment (cs-pooja-singh) [inline info.json]: we can remove this post dev

File: info.json [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+                "type": "checkbox",
+                "value": false,
+                "name": "load_threat"
+            }
+        ]
+    },
+    "operations": [
+        {
+            "operation": "bad_ip",
+            "title": "Bad IP",
```
Comment (cs-pooja-singh) [inline info.json]: Fetch a malicious IP Address

File: operations.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
+import json, requests, os, random, re, arrow
+from connectors.core.connector import get_logger, ConnectorError
+
+logger = get_logger('FortiSOARSocSimulator')
+
+def __threatdata_from_file(filename , params):
+    file_path = "{}/threat_intelligence/{}.txt".format(os.path.dirname(__file__), filename)
+    lines = open(file_path).read().splitlines()
+    if params.get('random'):
```
Comment (cs-pooja-singh) [inline operations.py]: why not pick random always instead of giving this as user input?

File: scenarios/Brute Force Attack/scenario_record.json [hunk 8]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (cs-pooja-singh) [inline scenarios/Brute Force Attack/scenario_record.json]: let's remove the .DS_Store files from the bundle

File: utils.py [hunk 9]
```diff
@@ -0,0 +1,175 @@
+#FSR Autogenerated Content. DO NOT DELETE
```
Comment (cs-pooja-singh) [inline utils.py]: can be removed in prod

File: utils.py [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+            method = "PUT"
+        elif result.status_code == 404:
+            url = "/api/3/scenario"
+            method = "POST"
+        else:
+            logger.error("Error getting record information.")
+            raise ConnectorError("Error getting record information.")
+
+        res = make_request(url, method, body=record)
+        logger.error(result.content)
```
Comment (cs-pooja-singh) [inline utils.py]: debug

File: utils.py [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+       
+        logger.error(scenario_title)
```
Comment (cs-pooja-singh) [inline utils.py]: debug

File: utils.py [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+                    f.write(response.content)
+
+        except Exception as e:
+            logger.error( "Error downloading threat intelligence data : {}".format(e) )
+            raise ConnectorError( "Error downloading threat intelligence data : {}".format(e) )
+
+
+
+
+def make_request(url, method, body=None,files=None, *args, **kwargs):
```
Comment (cs-pooja-singh) [inline utils.py]: call the function from the SDK instead

File: threat_intelligence/malicious_urls.txt [hunk 13]
```diff
@@ -0,0 +1,1593 @@
+http://bzmur9mdmpbzd5272cd.wisuda.ump.ac.id/
```
Comment (cs-pooja-singh) [inline threat_intelligence/malicious_urls.txt]: what is the source of this data? can we add an acknowledgement here?

File: connector.py (hunk 1 above)
Comment (cs-tushar-kanade) [reply connector.py]: Done

File: info.json (hunk 2 above)
Comment (cs-tushar-kanade) [reply info.json]: Done

File: info.json (hunk 3 above)
Comment (cs-tushar-kanade) [reply info.json]: done

File: info.json (hunk 4 above)
Comment (cs-tushar-kanade) [reply info.json]: done

File: info.json (hunk 5 above)
Comment (cs-tushar-kanade) [reply info.json]: done

File: info.json (hunk 6 above)
Comment (cs-tushar-kanade) [reply info.json]: done and updated for other entities

File: operations.py (hunk 7 above)
Comment (cs-tushar-kanade) [reply operations.py]: One of use case of scenario, I will check with Swati why it was provided and which use case it was

File: scenarios/Brute Force Attack/scenario_record.json (hunk 8 above)
Comment (cs-tushar-kanade) [reply scenarios/Brute Force Attack/scenario_record.json]: done

File: utils.py (hunk 9 above)
Comment (cs-tushar-kanade) [reply utils.py]: done

File: utils.py (hunk 10 above)
Comment (cs-tushar-kanade) [reply utils.py]: done

File: utils.py (hunk 11 above)
Comment (cs-tushar-kanade) [reply utils.py]: i removed those logger message as they were added to check during development

File: threat_intelligence/malicious_urls.txt (hunk 13 above)
Comment (cs-tushar-kanade) [reply threat_intelligence/malicious_urls.txt]: These were free at time when first version of connector was done, need to find free one for domain.

**Review by cs-pooja-singh — APPROVED** [review]

**Pull request merged**